jsm.go

```diff
@@ -259,34 +259,57 @@ type consumerResponse struct {
 
 // AddConsumer will add a JetStream consumer.
 func (js *js) AddConsumer(stream string, cfg *ConsumerConfig, opts ...JSOpt) (*ConsumerInfo, error) {
-	if cfg != nil && cfg.Durable != _EMPTY_ {
-		consInfo, err := js.ConsumerInfo(stream, cfg.Durable)
+	if cfg == nil {
+		cfg = &ConsumerConfig{}
+	}
+	if cfg.Name != _EMPTY_ && !js.nc.serverMinVersion(2, 9, 0) {
+		return nil, fmt.Errorf("%w: %s", ErrRequireServerVersion, "consumer name requires at least server version 2.9.0")
+	}
+	if cfg.Name != _EMPTY_ && cfg.Durable != _EMPTY_ && cfg.Name != cfg.Durable {
```

Comment:
The server already sends an error if sending to the new API endpoint but with a durable in the request that does not match the name in the subject. So not sure we have to do this in the client.

Reply:
I just updated the C PR with a test that uses different name and durable and verified that server sends an error message already.

Reply:
I wasn't sure whether we should also add client side validation for this case. I guess you're right, I'll remove it.

Reply:
It is always a choice between "saving on unnecessary network round-trip" and adding "duplicated" checks in the client library. Since server is doing the check already, I would remove from client.

Reply:
done!